### PR TITLE
Add `--force-platform` to `conda-lock install` command.

### PR DESCRIPTION
**Pull request by jezdez**
### Description

This selects a specific platform for installation instead of the current, native platform. Should be similar to https://github.com/conda/conda/pull/11794, kinda.

### CODE REVIEW

**Comment by maresb** [comment]
Thanks @jezdez! This makes sense.

Some initial thoughts:
1. We should add a test
2. I'm debating whether calling it the mundane `--platform` option is a good idea, since the use case is fairly obscure. Probably it's good for consistency, but I think we should at least indicate in the help string that the option is for forcing an override of the native platform, rather than something routine
3. For internal consistency, how about renaming `install(platform=...)` to `install(force_platform=...)`?

**Comment by jezdez** [comment]
@maresb Yup, this went to the org admins and I just approved it! (more on the PR tomorrow)

**Comment by jezdez** [comment]
> Thanks @jezdez! This makes sense.
> 
> Some initial thoughts:
> 
> 1. We should add a test

I've added one, reusing one of the lockfiles to mimic installing a Intel macOS package on the macOS ARM64 GitHub runners.

> 2. I'm debating whether calling it the mundane `--platform` option is a good idea, since the use case is fairly obscure. Probably it's good for consistency, but I think we should at least indicate in the help string that the option is for forcing an override of the native platform, rather than something routine

I've updated the description

> 3. For internal consistency, how about renaming `install(platform=...)` to `install(force_platform=...)`?

That's fine, I've updated both the argument name as well as the CLI option to make sure it's obvious that it's not quite the same as the `--platform` of the render and lock subcommands.

File: tests/test_conda_lock.py [hunk 1]
```diff
@@ -2650,6 +2650,54 @@ def test_fake_conda_env(conda_exe: str, conda_lock_yaml: Path):
...
+    if Path(conda_lock_yaml).exists():
+        logging.debug(
+            "lockfile contents: \n\n=======\n%s\n\n==========",
+            Path(conda_lock_yaml).read_text(),
+        )
+
+    assert _check_package_installed(
+        package=package,
+        prefix=str(prefix),
+        subdir=platform,
```
Comment (jezdez) [inline tests/test_conda_lock.py:2697]: It's checking the subdir value of the metadata here

**Comment by jezdez** [comment]
Thank you @maresb!

**Comment by leeleavitt** [comment]
When will this be released?

**Comment by maresb** [comment]
@leeleavitt, apologies for the delay, you can track the progress towards release in https://github.com/conda/conda-lock/issues/725.

**Pull request merged**